diff --git a/node_modules/vision-camera-resize-plugin/android/build.gradle b/node_modules/vision-camera-resize-plugin/android/build.gradle
index 88116f9..f0febf8 100644
--- a/node_modules/vision-camera-resize-plugin/android/build.gradle
+++ b/node_modules/vision-camera-resize-plugin/android/build.gradle
@@ -76,7 +76,9 @@ android {
       cmake {
         cppFlags "-O2 -frtti -fexceptions -Wall -fstack-protector-all"
         abiFilters (*reactNativeArchitectures())
-        arguments "-DANDROID_STL=c++_shared"
+        arguments "-DANDROID_STL=c++_shared",
+                  "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
+                  "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,max-page-size=16384"
       }
     }
   }
